base/subarray.jl

```diff
@@ -122,10 +122,16 @@ _maybe_reshape_parent(A::AbstractArray, ::NTuple{N, Bool}) where {N} = reshape(A
 """
     view(A, inds...)
 
-Like [`getindex`](@ref), but returns a view into the parent array `A` with the
-given indices instead of making a copy.  Calling [`getindex`](@ref) or
-[`setindex!`](@ref) on the returned `SubArray` computes the
-indices to the parent array on the fly without checking bounds.
+Like [`getindex`](@ref), but returns a lightweight array that lazily references
+(or is effectively a _view_ into) the parent array `A` at the given index or indices
+`inds` instead of eagerly extracting elements or constructing a copied subset.
+Calling [`getindex`](@ref) or [`setindex!`](@ref) on the returned value
+(often a [`SubArray`](@ref)) computes the indices to access or modify the
+parent array on the fly (without checking bounds a second time).
```

Comment:
`v[i]` checks indexing for the view itself so I think "without checking bounds" could be a bit confusing for new users. Maybe this clarifies the message?

```suggestion
parent array on the fly.  The behavior is undefined if the shape of the parent array is
changed after `view` is called because there is no bound check for the parent array; e.g.,
it may cause a segmentation fault.
```

Reply:
(I just realized that this is a phrase from the old version. But since we are cleaning things up, why not do it here as well?)

Reply:
Hmm I'm actually surprised this works:
```julia
julia> a = randn(6)
6-element Array{Float64,1}:
 -1.886744376748863
 -1.8200751803835573
  0.9900701536180667
  1.231029058457635
 -1.2857510124233005
  0.7893254374129456

julia> b = view(a,1:5)
5-element view(::Array{Float64,1}, 1:5) with eltype Float64:
 -1.886744376748863
 -1.8200751803835573
  0.9900701536180667
  1.231029058457635
 -1.2857510124233005

julia> resize!(a,3)
3-element Array{Float64,1}:
 -1.886744376748863
 -1.8200751803835573
  0.9900701536180667

julia> b
5-element view(::Array{Float64,1}, 1:5) with eltype Float64:
 -1.886744376748863
 -1.8200751803835573
  0.9900701536180667
  1.231029058457635
 -1.2857510124233005

julia> b[5]
-1.2857510124233005
```
I'm assuming that the memory isn't "freed" so this can't lead to memory access errors?

Reply:
Yeah, Julia's vector tracks the capacity (`maxsize`) and length separately. It looks like Julia decides to keep the old capacity in this case:

```
$ gdb -q --args usr/bin/julia-debug -q
Reading symbols from usr/bin/julia-debug...done.
(gdb) b jl_breakpoint
Function "jl_breakpoint" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (jl_breakpoint) pending.
(gdb) r
Starting program: /home/arakaki/repos/watch/julia/usr/bin/julia-debug -q
warning: Probes-based dynamic linker interface failed.
Reverting to original interface.

[New LWP 44378]
[LWP 44378 exited]
process 44367 is executing new program: /home/arakaki/repos/watch/julia/usr/bin/julia-debug
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff1a01700 (LWP 44379)]
[New Thread 0x7fffdf8f7700 (LWP 44380)]
[New Thread 0x7fffdf0f6700 (LWP 44381)]
[New Thread 0x7fffd68f5700 (LWP 44382)]
[New Thread 0x7fffce0f4700 (LWP 44383)]
[New Thread 0x7fffc58f3700 (LWP 44384)]
[New Thread 0x7fffbd0f2700 (LWP 44385)]
[New Thread 0x7fffb48f1700 (LWP 44386)]
julia> a = rand(6)
6-element Vector{Float64}:
 0.6799836266734105
 0.9026327305030659
 0.41122239092879687
 0.8319312383392878
 0.13829068479318196
 0.7567607359242641

julia> resize!(a, 3)
3-element Vector{Float64}:
 0.6799836266734105
 0.9026327305030659
 0.41122239092879687

julia> ccall(:jl_breakpoint, Cvoid, (Any,), a)

Thread 1 "julia-debug" hit Breakpoint 1, jl_breakpoint (v=0x7fffe5fad210)
    at /home/arakaki/repos/watch/julia/src/rtutils.c:1257
1257    {
(gdb) p ((jl_array_t *)v)->maxsize
$1 = 6
(gdb) p ((jl_array_t *)v)->length
$2 = 3
```